### PR TITLE
Feat/ocm shortcircuit migrate

### PR DESCRIPTION
**Pull request by ashishmax31**
# py-mtcli

## Description
Adds a new method called `shortcircuit_migrate` to the ocm client class which creates the migration object, enables it and then whitelists it, all wrapped in timeouts and max tries.

### CODE REVIEW

File: managedtenants/utils/general_utils.py [hunk 1]
```diff
@@ -23,3 +26,34 @@ def run(cmd, logger=None):
...
+
+    signal.signal(signal.SIGALRM, timeout_handler)
+    signal.alarm(duration)
+    try:
+        yield
+    finally:
+        signal.alarm(0)
+
+
+def try_with_timeout_until(
```
Comment (apahim) [inline managedtenants/utils/general_utils.py:44]: looks a lot like the sretoolbox.utils.retry https://github.com/app-sre/sretoolbox/blob/master/sretoolbox/utils/retry.py

Comment (ashishmax31) [reply managedtenants/utils/general_utils.py:44]: @apahim I'm not really sure the retry API provided in the sre toolbox is right here. The retry provided in the sre toolbox works on exceptions, whereas we want retry on predicates.

Comment (apahim) [reply managedtenants/utils/general_utils.py:44]: fair enough

File: managedtenants/utils/ocm.py [hunk 2]
```diff
@@ -162,6 +165,36 @@ def disable_addon(self, addon_id):
             f"/api/clusters_mgmt/v1/addons/{addon_id}", json={"enabled": False}
         )
 
+    def shortcircuit_migrate(self, addon_id):
+        def addon_migration_complete():
+            migration = self.get_addon_migration(addon_id).json()
+            return migration.get("state") == "completed"
```
Comment (yashvardhan-kukreja) [inline managedtenants/utils/ocm.py:171]:
```suggestion
        addon_migration_complete = self.get_addon_migration(addon_id).json().get("state") == "completed"
```
wouldn't this be cleaner.
I don't understand how this function specifically helps :)

Comment (yashvardhan-kukreja) [inline managedtenants/utils/ocm.py:171]:
```suggestion
        addon_migration_complete = lambda _: self.get_addon_migration(addon_id).json().get("state") == "completed"
```

imo we can use the benefits of lambda function here :)

Comment (ashishmax31) [reply managedtenants/utils/ocm.py:171]: I thought of using it as well. But the linter is not happy about the line length and as far as i know there is no way to split a lambda function into multiple lines in python. :/

Comment (yashvardhan-kukreja) [reply managedtenants/utils/ocm.py:171]: ahh gotcha, we could skip the linter check but ehh too much hassle.

**Review by yashvardhan-kukreja — APPROVED** [review]

**Pull request merged**